lib/jake.ex

```diff
@@ -11,46 +11,89 @@ defmodule Jake do
     "string"
   ]
 
-  def gen(%{"anyOf" => options} = spec) when is_list(options) do
+  def generator(map) do
+    StreamData.sized(fn size ->
+      Map.put(%{}, "map", map) |> Map.put("omap", map) |> Map.put("size", size) |> gen_init()
```

Comment:
can we use a proper struct here. Also use some name that's descriptive, map is the name of a datastructure. I don't understand what is omap, nmap, and map supposed to mean in all the places.

From what I can see, we need the root spec, the current position (some child element of the root spec) and size. But we seem to be mixing up spec, map etc everywhere.

Reply:
I've not used struct in elixir before, therefore it didn't click me. I just looked into `Structs` and it seems to be better choice here.

Here omap->`o`riginal root map, nmap-> `n`ew map which we are creating, map->sub-schema of root schema/map.
About variable names, I also thought they needed some change. 
Should we use something like `root_schema and child_schema` here?

> But we seem to be mixing up spec, map etc everywhere

I didn't get this point.

Reply:
> Should we use something like root_schema and child_schema here?
if we name the struct as Schema, then root/child sounds better

> But we seem to be mixing up spec, map etc everywhere
I mean, mostly about why do we need so many variables (spec, omap, nmap, map) etc, but in reality we only need root/child. Let's convert to struct first, then this issue will be solved.